app/actions/service_instance_update_user_provided.rb

```diff
@@ -30,6 +30,8 @@ def update(service_instance, message)
       service_instance.db.transaction do
         original_service_instance = service_instance.dup
         service_instance.save_with_new_operation(updates, last_operation)
+        update_service_bindings(service_instance, updates) if updates.key?(:credentials) || updates.key?(:syslog_drain_url)
+        update_route_bindings(service_instance, updates) if updates.key? :route_service_url
```

Comment:
[Minor] I would prefer a consistent check for existing entries in `updates`. Currently there are three (two and a half) different variants:
```
if updates.key?(:credentials)
if updates.key? :route_service_url
if updates[:syslog_drain_url]
```